tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,282 @@
+
+                youngest_first_sync_date = max(
+                    self.parse_date(record.get(replication_key))
+                    for record in first_sync_messages)
+
+                if len(resuming_sync_messages) > 0:  # 'collects' stream may be empty
+                    actual_oldest_resuming_replication_date = min(
+                        self.parse_date(record.get(replication_key))
+                        for record in resuming_sync_messages)
+
+                if stream in new_state['bookmarks'].keys():
+                    expected_resuming_sync_start_time = self.parse_date(first_bookmark_value)
+                else:
+                    # For not yet started streams there will be no bookmark and we should
+                    # sync all records the from the beginning of the original sync.
+                    expected_resuming_sync_start_time = min(
+                        self.parse_date(record.get(replication_key))
+                        for record in first_sync_messages)
+
+                # verify the syncs sets a bookmark of the expected form
+                self.assertIsNotNone(first_bookmark_value)
+                self.assertTrue(self.is_expected_date_format(first_bookmark_value))
+                self.assertIsNotNone(resuming_bookmark_value)
+                self.assertTrue(self.is_expected_date_format(resuming_bookmark_value))
+
+                # verify the resuming bookmark is greater than 1st sync bookmark
+                """
+                This is the expected behaviour for shopify as they are using date windowing
+                TDL-17096 : Resuming bookmark value is getting assigned from execution time rather
+                than the actual bookmark time for some streams.
+                TODO this is not the case for the transactions stream, they are equal.
+                """
+                self.assertGreaterEqual(resuming_bookmark_value, first_bookmark_value)
+
+                # verify oldest record from resuming sync respects bookmark from previous sync
+                if resuming_sync_messages and stream in new_state['bookmarks'].keys():
+                    self.assertEqual(actual_oldest_resuming_replication_date,
+                                     self.parse_date(simulated_bookmark_value))
+
+                    # all interrupted recs are in full recs, interrupted rec counts verified
+                    first_sync_records_after_bookmark = [
+                        record for record in first_sync_messages
+                        if self.parse_date(record[replication_key]) >=
+                        self.parse_date(simulated_bookmark_value)]
+                    # remove any records that got added after the first sync
+                    filtered_resuming_records = [
+                        record for record in resuming_sync_messages
+                        if self.parse_date(record[replication_key]) <=
+                        youngest_first_sync_date]
```

Comment:
> remove any records that got added after the first sync

Wouldn't this be the `oldest_first_sync_date` then? The `youngest_first_sync_date` should be the start date

Reply:
Actually I'm confused again after reading the assert right below this.

It seems like the intent is that the second sync should have the same records as the first sync once you account for a simulated bookmark

```
        first sync:  1 2 3 4 5 6 7 8 9
simulated bookmark:          5
       second sync:          5 6 7 8 9
```

So my confusion is why are we filtering the resuming sync at all? But if we are filtering, why aren't we filtering by the simulated bookmark?

Reply:
I agree, it's confusing.  Oldest date should be first record after start date.  Youngest should be last record in the sync.  The simulated bookmarks are lined up with record times.  Resuming sync filters out any data that may have come in between syncs so we can compare apples to apples and have the same end date for both syncs.